packages/plugin/src/PivotPlugin.js

```diff
@@ -1,192 +1,108 @@
+        // only add parameter if value !== default
         if (
-            this.props.id !== prevProps.id ||
-            !isEqual(this.props.style, prevProps.style)
+            visualization[option] !== undefined &&
+            visualization[option] !== props.defaultValue
         ) {
-            this.recreateVisualization(0) // disable animation
-            return
+            map[option] = visualization[option]
         }
-    }
-
-    getRequestOptions = (visualization, filters) => {
-        const options = getOptionsForRequest().reduce(
-            (map, [option, props]) => {
-                // only add parameter if value !== default
-                if (
-                    visualization[option] !== undefined &&
-                    visualization[option] !== props.defaultValue
-                ) {
-                    map[option] = visualization[option]
-                }
 
-                return map
-            },
-            {}
-        )
+        return map
+    }, {})
 
-        // interpretation filter
-        if (filters.relativePeriodDate) {
-            options.relativePeriodDate = filters.relativePeriodDate
-        }
-
-        // global filters
-        // userOrgUnit
-        if (filters.userOrgUnit && filters.userOrgUnit.length) {
-            const ouIds = filters.userOrgUnit.map(
-                ouPath => ouPath.split('/').slice(-1)[0]
-            )
-
-            options.userOrgUnit = ouIds.join(';')
-        }
-
-        return options
+    // interpretation filter
+    if (filters.relativePeriodDate) {
+        options.relativePeriodDate = filters.relativePeriodDate
     }
 
-    renderTable = async () => {
-        const {
-            config: visualization,
-            filters,
-            onResponsesReceived,
-            onError,
-        } = this.props
-
-        const i18nManager = {
-            get: string => i18n.t(string),
-        }
-
-        const appManager = {
-            getLegendSetById: () => '',
-            getApiPath: () => '',
-        }
+    // global filters
+    // userOrgUnit
+    if (filters.userOrgUnit && filters.userOrgUnit.length) {
+        const ouIds = filters.userOrgUnit.map(
+            ouPath => ouPath.split('/').slice(-1)[0]
+        )
 
-        const uiManager = {}
+        options.userOrgUnit = ouIds.join(';')
+    }
 
-        const d2aOptionConfig = new d2aConfig.OptionConfig()
-        d2aOptionConfig.setI18nManager(i18nManager)
-        d2aOptionConfig.init()
+    return options
+}
 
-        const refs = {
-            api: d2aApi,
-            appManager,
-            uiManager,
-            i18nManager,
-            optionConfig: d2aOptionConfig,
-        }
+const PivotPlugin = ({
+    config: visualization,
+    filters,
+    style,
+    onError,
+    onResponsesReceived,
+    id,
+    d2,
+}) => {
+    const [isLoading, setIsLoading] = useState(true)
+    const [data, setData] = useState(null)
+
+    const remappedOptions = {
```

Comment:
I think we can get rid of this, or is the new engine using some Layout stuff from the old PT app?

Reply:
It's not currently reading from the `visualization` object directly, though it could and then we'd have only a couple options left (like the clipping limit)

Reply:
Not using anything from the old app

Reply:
Ok, so no reason for renaming options then.